components/bind/Solaris/dns-server.sh

```diff
@@ -99,12 +100,54 @@ case "$method" in
 	    set -- `echo ${value} | /usr/bin/sed -e  's/\\\\//g'`
 	    server=$@
 	    ;;
+	'user')
+	    cmdopts="${cmdopts} -u ${value}"
+	    cmduser=${value};
+	    ;;
 	esac
     done
 
+    # Check configuration file exists.
+    if [ ! -f ${configuration_file} ]; then
+      msg="$I : Configuration file ${configuration_file} does not exist!"
+      echo ${msg} >&2
+      /usr/bin/logger -p daemon.error ${msg}
+      # dns-server should be placed in maintenance state.
+      result=${SMF_EXIT_ERR_CONFIG}
+    fi
+
     # If chroot option is set, note zones(5) are preferred, then
     # configuration file lives under chroot directory.
     if [ "${chroot_dir}" != "" ]; then
```

Comment:
I like this, but please, add also check for $chroot_dir != / , to disallow user to shoot himself in the foot.